### PR TITLE
Added openarray[char] overloads to std/unicode

### PR DESCRIPTION
**Pull request by beef331**
Had to add a `substr` overload for `openArray` which enables the limiting behaviour.

### CODE REVIEW

File: lib/system.nim [hunk 1]
```diff
@@ -2592,6 +2592,27 @@ proc substr*(s: openArray[char]): string =
   for i, ch in s:
     result[i] = ch
 
+proc substr*(s: openArray[char] , first, last: int): string =
```
Comment (Araq) [inline lib/system.nim]: This is not used afaict.

**Comment by Araq** [comment]
Nice but the forwarding procs that take `string` should be marked with `inline`.

File: lib/system.nim (hunk 1 above)
Comment (beef331) [reply lib/system.nim]: `runeSubstr` uses it once, so seemed sensible to port it over to openarray, for an instance where someone wants the same behaviour for openarray in the future.

Comment (Araq) [reply lib/system.nim]: So move it to unicode.nim and don't export it for the time being.

**Pull request merged**